- Added ChipsView & ChipsDetailedView

dekaustubh opened the pull request:
- Some specific changes for Astro

Commit:
- Added ChipsView & ChipdsDetailedView
- Some specific changes for Astro

dekaustubh commented:
Advantages of using the same library is we don't need to change large amount of code in our astro App. Right now it looks like only few lines of change would work. Here is summary of this commit.

For now I have added chips view & chips details view. Both are configurable. The +n functionality was already in place so there was no need to change of that. Pasting also works, but it needs some more tweaking. 

TODO : Only one major change remains, which is creating a chip once use pastes email address or when user type email address instead of selecting one from the drop down. I am now moving onto these two changes.

Commit:
- Added support for pasting an email address

Commit:
- Made detailed chip view theme configurable

awhlee commented on example/src/main/java/com/tokenautocomplete/TokenActivity.java:
```diff
@@ -9,48 +9,51 @@
-                ((TextView)convertView.findViewById(R.id.email)).setText(p.getEmail());
+                ChipInterface p = getItem(position);
+                ((TextView)convertView.findViewById(R.id.name)).setText(p.getEmailAddress());
+                ((TextView)convertView.findViewById(R.id.email)).setText(p.getPhoneNumber());
```
phone number?

awhlee commented on example/src/main/java/com/tokenautocomplete/TokenActivity.java:
```diff
@@ -9,48 +9,51 @@
 
             @Override
-            protected boolean keepObject(Person person, String mask) {
+            protected boolean keepObject(ChipInterface person, String mask) {
```
rename person as to not confuse the person reading the code.

awhlee commented on example/src/main/java/com/tokenautocomplete/TokenActivity.java:
```diff
@@ -59,18 +62,11 @@ protected boolean keepObject(Person person, String mask) {
         completionView.setTokenClickStyle(TokenCompleteTextView.TokenClickStyle.Select);
 
-
-        if (savedInstanceState == null) {
```
Why did you remove this?  Please add more comments to your commits.  At some point we might need to rebase this against the original project.  So documenting where things changed is helpful at times

dekaustubh replied:
This is a sample activity. Just for demo. Doesn't affect the library. So these lines add the chips onCreate of an activity, I didn't want that in the demo to test all the cases.

awhlee replied:
for a third-party library best approach would be to comment it out and leave it there with a comment about why it is commented out.  this makes future rebases clearer.

awhlee commented on library/build.gradle:
```diff
@@ -31,6 +31,9 @@ task jar(type: Jar) {
 
 dependencies {
     compile 'com.android.support:support-annotations:23.3.0'
+    compile 'com.android.support:appcompat-v7:25.3.0'
```
is there a problem with having different versions of support libraries in the same project?  Seems like keeping them in sync is the better way to go.

awhlee commented on library/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,22 @@
+    </array>
+
+    <!-- The default letter tile text size -->
+    <dimen name="tile_letter_font_size">17sp</dimen>
```
these should not be in strings.xml

awhlee commented on library/src/main/res/values/colors.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <color name="colorChipViewBackground">#E0E0E0</color>
```
do these colors follow our designs?

awhlee replied:
if so, you might want to comment which the RGB maps to e.g. astroViolet500

awhlee reviewed (changes requested):
I'm going to put this current review on hold to wait for another update.  Need comments on what was pulled from the other library vs. what was written (and needs to be reviewed).

Commit:
- Fixed support for typing & pasting
- Added some more comments
- Fixed potential crashes
- Added Astro specific colors

Commit:
- Changed all support library versions to 25.3.1

Commit:
- Moved object list from Serializable to Parcelable

Commit:
- Updated README.md
